common/src/main/java/com/wynntils/features/user/map/MapFeature.java

```diff
@@ -81,4 +104,66 @@ public class MapFeature extends UserFeature {
     public List<Class<? extends Model>> getModelDependencies() {
         return List.of(MapModel.class);
     }
+
+    @SubscribeEvent
+    public void onRightClick(PlayerInteractEvent.RightClickBlock event) {
+        if (!autoWaypointChests) return;
+
+        lastChestPos = event.getPos();
+    }
+
+    @SubscribeEvent
+    public void onScreenOpened(ScreenOpenedEvent event) {
+        if (!autoWaypointChests) return;
+        if (lastChestPos == null) return;
+        if (!(event.getScreen() instanceof ContainerScreen)) return;
+
+        Matcher matcher = WynnScreenMatchers.lootChestMatcher(event.getScreen());
+        if (!matcher.matches()) return;
+
+        ChestTier tier = ChestTier.fromString(matcher.group(1));
+
+        if (tier.ordinal() < minTierForAutoWaypoint.ordinal()) return;
+
+        MapLocation location = new MapLocation(lastChestPos.getX(), lastChestPos.getY(), lastChestPos.getZ());
+        CustomPoi newPoi = new CustomPoi(
+                location, tier.getWaypointName(), CommonColors.WHITE, tier.getWaypointTexture(), Integer.MIN_VALUE);
+
+        if (MapFeature.INSTANCE.customPois.stream().noneMatch(customPoi -> customPoi.equals(newPoi))) {
```

Comment:
I wonder if we could store chestPois as a hashset separately.

Reply:
Too much complication imo

Reply:
It would be better for importing/exporting chest pois maybe if they were held separately

Reply:
We could filter for them ig. I don't wanna separate POIs into multiple sets. Maybe later we could have some kind of poi metadata to indicate it